build_tools/scripts/lint.sh

```diff
@@ -139,6 +139,13 @@ echo "***** Generates CMake files *****"
 git add -A
 git diff HEAD --exit-code
 
+echo "***** Check BUILD files are not named BUILD (prefer BUILD.bazel) *****"
+if [[ $(git ls-files '**BUILD') ]]; then
```

Comment:
You have to do the whole `enable_update_ret`, `disable_update_ret` thingy here, otherwise this will fail if there are no BUILD files and that will get captured as the return code and lint.sh will fail (run it on your PR and it will fail). I regret that I tried to mess with bash exit codes here. This whole thing should be a Python script and we should probably just invoke the Python script in a single GitHub action...

Reply:
I did test it. `git ls-files` actually exits 0 whether there are matches or not. But I added the disable/enable to make it more clear what we care about.

Reply:
Oh weird. I checked this out and ran it locally and got a failure, but I guess I mis-identified the cause. I missed that you've got `[[ ... ]]` wrapping the command here, so we're testing its stdout, not its exit code. In that case, you don't have to add the disable/enable. If the command failed for some reason, I think we would want it to get captured.

Reply:
SG - I removed the disable/enable so it checks all the commands now.